src/components/pages/GenerationPage/PetitionList.js

```diff
@@ -79,7 +131,43 @@ function PetitionRow({ attorney, petition, petitionerData, validateInput }) {
             ))}
           </Attachments>
         </TableCell>
+        <TableCell>
+          <GenerateButton 
+            label={isDetailed ? <FontAwesomeIcon icon={faChevronDown}/> : <FontAwesomeIcon icon={faChevronRight}/>}
+            isCollapsed=<FontAwesomeIcon icon={faChevronDown}/>
+            onClick={() => handlePress()}
+            title="Reveal offense records"
+          />
+        </TableCell>
       </TableRow>
+      {isDetailed && (
+        <TableRow key={petition.pk + 'offenses'} backgroundColor={backgroundColor}>
+          <TableSpanCell className='spanCell'>
+            { offenseRecordsLoading ? <h5>Loading...</h5> : (
+              <Table >
+                <TableBody>
+                  <TableRow key={petition.pk + '_offenses'}>
+                    <TableCell>
+                        <HighlightTable 
```

Comment:
This doesn't look right to me. Why are there three nested tables here? I think we can simplify:

```
<TableSpanCell>
  <FlexColumnDiv>
    <HighlightTable />
    <AlignSelfEndGenerateButton />
  </FlexColumnDiv>
</TableSpanCell>
```

Note: you'll need to add `align-self: end` or something like that to the generate button and you'll need to add a wrapper element with `display: flex` and `flex-flow: column`